public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx

```diff
@@ -33,59 +31,38 @@ export interface CloudWatchLogsQueryFieldProps
     Themeable2 {
   absoluteRange: AbsoluteTimeRange;
   onLabelsRefresh?: () => void;
-  ExtraFieldElement?: ReactNode;
+  ExtraFieldElement?: React.ReactNode;
   exploreId: ExploreId;
   query: CloudWatchLogsQuery;
 }
 
-const addPaddingToButton = css`
-  padding: 1px 4px;
-`;
-interface State {
-  hint:
-    | {
-        message: string;
-        fix: {
-          label: string;
-          action: () => void;
-        };
-      }
-    | undefined;
-}
-
-class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogsQueryFieldProps, State> {
-  state: State = {
-    hint: undefined,
-  };
-
-  plugins: Array<Plugin<Editor>>;
+export const CloudWatchLogsQueryField = (props: CloudWatchLogsQueryFieldProps) => {
+  const { query, datasource, onChange, onRunQuery, ExtraFieldElement, data } = props;
 
-  constructor(props: CloudWatchLogsQueryFieldProps, context: React.Context<any>) {
-    super(props, context);
+  const showError = data && data.error && data.error.refId === query.refId;
+  const cleanText = datasource.languageProvider ? datasource.languageProvider.cleanText : undefined;
 
-    this.plugins = [
-      BracesPlugin(),
-      SlatePrism(
-        {
-          onlyIn: (node: Node) => node.object === 'block' && node.type === 'code_block',
-          getSyntax: (node: Node) => 'cloudwatch',
-        },
-        { ...(prismLanguages as LanguageMap), cloudwatch: syntax }
-      ),
-    ];
-  }
-
-  componentDidMount = () => {
-    const { query, datasource, onChange } = this.props;
-
-    if (onChange) {
+  const plugins: Array<Plugin<Editor>> = [
+    BracesPlugin(),
+    SlatePrism(
+      {
+        onlyIn: (node: Node) => node.object === 'block' && node.type === 'code_block',
+        getSyntax: (node: Node) => 'cloudwatch',
+      },
+      { ...(prismLanguages as LanguageMap), cloudwatch: syntax }
+    ),
+  ];
+
+  React.useEffect(() => {
+    if (onChange && !query.logGroupNames) {
```

Comment:
I don't think onChange is optional, so you don't need to check if it defined. Don't know why it did so before.